fix(2d): call spawnChildren() in constructor to initialize spawner parents

exdeejay opened the pull request:
Fixes #270 by calling the spawner function once during construction, to initialize the parents of the spawner objects. A warning in the docs about making sure to null check the parent of spawners would be prudent as well.

aarthificial reviewed (changes requested):
This only fixes one side effect and not the core issue.
The problem will come back when you change the signal that the spawner depends on:
```ts
view.add(<Array ref={ArrayRef} values={[0, 1]} />);
ArrayRef().values([0, 1, 2, 3, 4]); // happens after the constructor

yield* ArrayRef().Swap(1, 2);
```

The real issue here lies in the user land. When using spawners with a pool, you shouldn't access the pool directly. Instead, you should access the nodes thorough the `children` property of the spawner node:
```ts
export class Array {
  // ...
  
  // a reference to the spawner node
  public boxes: Node;

  // access the boxes through its `children`
  public get boxArray() {
    return this.boxes.children() as Rect[];
  }

  public constructor(props?: ArrayProps) {
    super({
      ...props,
    });

    this.add(
      <Node
        ref={makeRef(this, 'boxes')}
        spawner={() => this.pool.slice(0, this.values().length)}
      ></Node>,
    );
  }
}
```
This will make sure that everything works correctly.

Adding `this.spawnChildren()` to the base constructor comes with some issues on its own (can't use `this` in the spawner function, for example). 
It may also result in people developing bad habits of accessing objects from the pool. 

Let's add an explanation of this to the documentation instead.

exdeejay commented:
Reverted the above change and added guidance to use `children()` wherever possible. I'm not too familiar with how spawners work and there aren't any public examples I can find, so I can't add more guidance than that.
